studyzen/Lectures/LectureService.cs

```diff
@@ -0,0 +1,72 @@
+using Studyzen.Lectures;
+using StudyZen.Lectures.Requests;
+using StudyZen.Persistence;
+using StudyZen.Utils;
+
+namespace StudyZen.Lectures;
+
+public interface ILectureService
+{
+    Lecture AddLecture(CreateLectureRequest request);
+    Lecture? GetLectureById(int lectureId);
+    IEnumerable<Lecture> GetLecturesByCourseId(int? courseId);
+    Lecture? UpdateLectureById(int lectureId, string? name, string? content);
```

Comment:
I suggest passing the whole request object. This way we can validate the object if we decide to add validation (https://docs.fluentvalidation.net/en/latest/). Also, easier to maintain and add new fields

Reply:
fixed for parameters from request body, but not for query params as it is kind of useless to create a whole new class for one id